integration_tests/test_benchmark.py

```diff
@@ -26,18 +26,14 @@ def test_bench():
         print(stderr)
         assert return_code == 0
 
-    # Move to upstream tip.
-    subprocess.run(
-        "git fetch {} {}".format(REMOTE, BRANCH),
-        shell=True, check=True
-    )
-    subprocess.run(
-        "git checkout {}/{}".format(REMOTE, BRANCH),
-        shell=True, check=True
-    )
-
     # Get numbers from upstream tip, without the changes from the current PR.
+    _checkout_master_branch()
     return_code, stdout, stderr = _run_cargo_bench("before")
+
+    # Before checking any results, let's just go back to the PR branch.
```

Comment:
Is is guaranteed that tests won't run in parallel for the same source folder?

Reply:
There cannot be two tests running on the same directory at the same time. One agent can run maximum one test at the time. If there are multiple agents running tests, than each of them will have their own directory in which to check out code.